### PR TITLE
Implement "earlyMessageRejectionPeriod" config discussed in issue #4599

### PR DESCRIPTION
**Pull request by wywlds**
#4599 
    * implemented the logics of this param
    * Added doc of this config
    * Added unit tests of it

### CODE REVIEW

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java [hunk 1]
```diff
@@ -958,9 +963,9 @@ private void addDiscoveredTaskToPendingCompletionTaskGroups(
 
     log.info("Creating new pending completion task group for discovered task [%s]", taskId);
 
-    // reading the minimumMessageTime from the publishing task and setting it here is not necessary as this task cannot
+    // reading the minimumMessageTime&maximumMessageTime from the publishing task and setting it here is not necessary as this task cannot
```
Comment (jihoonson) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: ```minimumMessageTime & maximumMessageTime``` looks better to read.

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java [hunk 2]
```diff
@@ -1307,7 +1312,7 @@ private void checkCurrentTaskState() throws ExecutionException, InterruptedExcep
 
       // Iterate the list of known tasks in this group and:
       //   1) Kill any tasks which are not "current" (have the partitions, starting offsets, and minimumMessageTime
-      //      (if applicable) in [taskGroups])
+      //      &maximumMessageTime (if applicable) in [taskGroups])
```
Comment (jihoonson) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: Same here.

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java (hunk 1 above)
Comment (wywlds) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: Hi I changed as what you suggest.

Comment (wywlds) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: Thank you for the review! 
I already fixed the conflict.

**Comment by jihoonson** [comment]
@wywlds thank you for the quick fix. There are some format errors. Would you fix it too?

**Comment by wywlds** [comment]
@jihoonson I fixed the format errors, thank you!

**Comment by jihoonson** [comment]
I'm closing and reopening this pr to trigger travis.

**Comment by jihoonson** [comment]
@wywlds there are still format errors. Would you please check again?

**Comment by wywlds** [comment]
@jihoonson I think all checks passed this time.

**Pull request merged**